From: Ayman <aymn@pipa>
Date: Mon, 15 Jun 2026 21:00:00 +0000
Subject: [PATCH 10/10] media: hi846: Set clock to 24MHz for Xiaomi Pad 6 front camera

The Snapdragon 865 (SM8250) camera clock controller rejects 25MHz requests,
falling back to the 19.2MHz base clock. This causes the front camera stream
to hang due to link frequency mismatch. Switch to 24MHz which is natively
supported by the CAMCC hardware.

Signed-off-by: Ayman <aymn@pipa>

--- a/arch/arm64/boot/dts/qcom/sm8250-xiaomi-pipa.dts
+++ b/arch/arm64/boot/dts/qcom/sm8250-xiaomi-pipa.dts
@@ -825,9 +825,9 @@
 		vddio-supply = <&vreg_l1c_1p8>;
 
 		clocks = <&camcc CAM_CC_MCLK3_CLK>;
-		clock-frequency = <25000000>;
+		clock-frequency = <24000000>;
 		assigned-clocks = <&camcc CAM_CC_MCLK3_CLK>;
-		assigned-clock-rates = <25000000>;
+		assigned-clock-rates = <24000000>;
 
 		reset-gpios = <&tlmm 109 GPIO_ACTIVE_LOW>;
 		pinctrl-0 = <&mclk3_active &camera_front_active>;
--- a/drivers/media/i2c/hi846.c
+++ b/drivers/media/i2c/hi846.c
@@ -3214,7 +3214,7 @@ static int hi846_probe(struct i2c_client *client)
 	}
 
 	mclk_freq = clk_get_rate(hi846->clock);
-	if (mclk_freq != 25000000) {
+	if (mclk_freq != 24000000) {
 		dev_err(&client->dev, "External clock freq %u is non-standard\n",
 			mclk_freq);
 		return -EINVAL;
